### PR TITLE
Lighter SQL requests and compat with Oracle

### PR DESCRIPTION
**Pull request by ArtificialOwl**

### CODE REVIEW

**Comment by ArtificialOwl** [comment]
fix #901
fix #789 

This is a huge PR, split in multiple commit:

1. **import of MSPT**: the external tools lib is now fully copied within the app, using `OCA\Circles\Tools` as namespace. 
2. **switch import to OCA\Circles\Tools**: calls to MSPT have been changed to reflect the import of the tools lib.
3. **population stored as setting**: the idea here is to remove the calculation of the population of Circles during the SELECT request. Instead, the population is stored as a Circle's Setting. The calculation will be added in another Pull Request, until then population will be displayed at 0.
4. **removing useless SELECT DISTINCT and GROUPBY**.
5. **instance can be empty string**: Oracle does not like `$expr->eq()` on empty string, switching to `$expr->emptyString()` when needed.

File: lib/Controller/RemoteController.php [hunk 1]
```diff
@@ -534,4 +534,76 @@ public function exceptionResponse(
...
+	 *      return new DataResponse(['test' => 42]);
+	 * } catch (JsonNotRequestedException $e) {}
+	 *
+	 *
+	 * @throws NotLoggedInException
+	 * @throws JsonNotRequestedException
+	 */
+	private function publicPageJsonLimited(): void {
+		if (!$this->jsonRequested()) {
+			if (!OC::$server->get(IUserSession::class)
```
Comment (CarlSchwan) [inline lib/Controller/RemoteController.php:553]: Dependency injection

File: lib/Controller/RemoteController.php [hunk 2]
```diff
@@ -534,4 +534,76 @@ public function exceptionResponse(
...
+	}
+
+
+	/**
+	 * @param array $needles
+	 *
+	 * @return bool
+	 */
+	private function areWithinAcceptHeader(array $needles): bool {
+		$request = OC::$server->get(IRequest::class);
```
Comment (CarlSchwan) [inline lib/Controller/RemoteController.php:583]: Dependency injection

File: lib/Controller/RemoteController.php [hunk 3]
```diff
@@ -534,4 +534,76 @@ public function exceptionResponse(
...
+		foreach ($accepts as $accept) {
+			if (in_array($accept, $needles)) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
```
Comment (CarlSchwan) [inline lib/Controller/RemoteController.php:595]: php doc comments that just repeat the type information of the method are useless

**Review by CarlSchwan — APPROVED** [review]

**Pull request merged**